client/src/overview/subComponents/ImageGallery.jsx

```diff
@@ -5,41 +5,73 @@ import { StyleIdxContext } from '../overview.jsx';
 export default function ImageGallery({ currentProductStyles }) {
   const StyleIdxContextData = useContext(StyleIdxContext);
   const stylePics = [];
+  const previewPics = [];
   const [currentSlide, setCurrentSlide] = useState(0);
-
+  const [currentPreviewIdx, setCurrentPreviewIdx] = useState(0);
   let slideLength = 0;
   let currentStyleIdx;
   if (StyleIdxContextData && currentProductStyles) {
     ({ currentStyleIdx } = StyleIdxContextData);
     currentProductStyles.results[currentStyleIdx].photos.forEach((photo) => {
       stylePics.push(photo.url);
     });
+    currentProductStyles.results[currentStyleIdx].photos.forEach((photo) => {
+      previewPics.push(photo.thumbnail_url);
+    });
     slideLength = stylePics.length;
   }
 
   const changeSlide = (event, handle) => {
     if (handle === 'next') {
       setCurrentSlide(currentSlide === slideLength - 1 ? 0 : currentSlide + 1);
+      setCurrentPreviewIdx(currentSlide === slideLength - 1 ? 0 : currentSlide + 1);
     } else if (handle === 'prev') {
       setCurrentSlide(currentSlide === 0 ? slideLength - 1 : currentSlide - 1);
+      setCurrentPreviewIdx(currentSlide === 0 ? slideLength - 1 : currentSlide - 1);
     }
   };
 
+  const previewClicked = (event, index) => {
+    setCurrentPreviewIdx(index);
+    setCurrentSlide(index);
+  };
+
   useEffect(() => {
     setCurrentSlide(0);
+    setCurrentPreviewIdx(0);
   }, [currentStyleIdx]);
 
   return (
     <div className='imageGallery'>
+      <div className='previewContainer'>
+      { currentSlide === 0
+        ? null
+        : <button className={ 'fas fa-chevron-up up-arrow vertical-arrow'} onClick={ () => changeSlide(event, 'prev') } />}
```

Comment:
Maybe we should adjust the format of the code here? At first glance, it looks like the previewPics elements are nested inside of the logic for the arrow. iIt might be as simple as moving the last `}` onto its own line